script/asciidoctor-extensions.rb

```diff
@@ -0,0 +1,63 @@
+require 'asciidoctor'
+require 'asciidoctor/extensions'
+require 'asciidoctor/converter/html5'
+
+module Git
+  module Documentation
+    class SynopsisBlock < Asciidoctor::Extensions::BlockProcessor
+
+      use_dsl
+      named :synopsis
+      parse_content_as :simple
+
+      def process parent, reader, attrs
+        outlines = reader.lines.map do |l|
+          l.gsub(/(\.\.\.?)([^\]$.])/, '`\1`\2')
```

Comment:
I think a line of comment wouldn't hurt with these regexes. Maybe best with an example:

```suggestion
          l.gsub(/(\.\.\.?)([^\]$.])/, '`\1`\2') # wrap ellipsis in backticks: ...something => `...`something
```

I think the intended use is for `[...<more>]`? Should we include the `[` and `]` in the regex?

Reply:
This line is trying to differentiate the three dots in different contexts, where they have a different meaning and require different formatting.

First there is the form \<commit1\>...\<commit2\> when describing a range of commits, where the three dots are a "keyword" understood by git and must be formatted as code.

Then there is the forms used in the grammar to express repetition, such as in "<path> ..." with optionally square brackets, such as "[<path>...]" which usually appear at the end of the command line. These three dots must not be formatted as code, but left as is.

This line matches the former case and forces the corresponding format. I'll add  a comment in the same line as yours.